js/parser.js

```diff
@@ -351,6 +369,74 @@ var ERMrest = (function(module) {
         },
 
         /**
+         * string of the search filter in the URI
+         * @returns {string}
+         */
+        get searchFilter() {
+            return this._searchFilter;
+        },
+
+        /**
+         * Apply, replace, clear filter term on the location
+         * @param {string} term - optional, set or clear search
+         */
+        search: function(term) {
+
+            var filterString = "";
+
+            if (term && term !== "") {
+                this._searchTerm = term;
+
+                // convert term to filter
+
+                // 1) parse terms in quotation
+                // 2) split the rest by space
+                var terms = term.match(/"[^"]*"/g); // everything that's inside quotation
+                if (!terms) terms = [];
+                for (var i = 0; i < terms.length; i++) {
+                    term = term.replace(terms[i], ""); // remove from term
+                    terms[i] = terms[i].replace(/"/g, ""); //remove quotes
+                }
+
+                terms = terms.concat(term.trim().split(/[\s]+/)); // split by white spaces
+
+                terms.forEach(function(t, index, array) {
+                    filterString += (index === 0? "" : "&") + "*::ciregexp::" + module._fixedEncodeURIComponent(t);
+                });
+
+            } else {
+                this._searchTerm = null;
+            }
+
+            if (this._searchFilter && term) { // replace search filter
```

Comment:
These if statements are a bit confusing.

```javascript
if (this._searchFilter) {
    if (filterString) { // replace search filter

    } else { // remove search filter 

    }
} else if (filterString) { // add search filter

}  
```

This looks easier to understand to me.